### PR TITLE
IMPRO-1434: Change Return types to follow docstring style guide.

### PR DESCRIPTION
**Pull request by neilCrosswaite**
Changed docstrings so that the Return type is no longer in brackets and the variable name has been removed.

Tidied up other docstrings that i found along the way.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: lib/improver/cli/temporal_interpolate.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def process(cube_0, cube_1, interval_in_mins=None, in_times=None,
             Default is linear.
 
     Returns:
-        result (iris.cube.Cubelist):
+        iris.cube.Cubelist:
```
Comment (cgsandford) [inline lib/improver/cli/temporal_interpolate.py]: Should this be `iris.cube.CubeList`?

File: lib/improver/wxcode/weather_symbols.py [hunk 2]
```diff
@@ -398,7 +398,7 @@ def find_all_routes(graph, start, end, route=None):
                 A list of node names found so far.
 
         Returns:
-            routes (list):
+            list:
```
Comment (cgsandford) [inline lib/improver/wxcode/weather_symbols.py]: `list of str:`

File: lib/improver/between_thresholds.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def _slice_cube(self):
         Extract required slices from input cube
 
         Returns:
-            cubes (list):
+            list:
```
Comment (cgsandford) [inline lib/improver/between_thresholds.py]: `list of iris.cube.Cube:`

File: lib/improver/blending/weights.py [hunk 4]
```diff
@@ -387,7 +387,7 @@ def _define_slice(self, cube):
                 Cube input to plugin
 
         Returns:
-            slice_list (list):
+            list:
```
Comment (cgsandford) [inline lib/improver/blending/weights.py]: `list of iris.coords.Coord:`

File: lib/improver/ensemble_calibration/ensemble_calibration.py [hunk 5]
```diff
@@ -155,7 +155,7 @@ def process(
                 minimisation within self.minimisation_dict.
 
         Returns:
-            optimised_coeffs (list):
+            list:
```
Comment (cgsandford) [inline lib/improver/ensemble_calibration/ensemble_calibration.py]: `list of float:`

File: lib/improver/ensemble_calibration/ensemble_calibration.py [hunk 6]
```diff
@@ -646,7 +646,7 @@ def compute_initial_guess(
                 used as predictors. Default is None.
 
         Returns:
-            initial_guess (list):
+            list:
```
Comment (cgsandford) [inline lib/improver/ensemble_calibration/ensemble_calibration.py]: `list of float:`

File: lib/improver/ensemble_copula_coupling/ensemble_copula_coupling_utilities.py [hunk 7]
```diff
@@ -94,7 +94,7 @@ def choose_set_of_percentiles(no_of_percentiles, sampling="quantile"):
             * Random: A random set of ordered percentiles.
 
     Returns:
-        percentiles (list):
+        list:
```
Comment (cgsandford) [inline lib/improver/ensemble_copula_coupling/ensemble_copula_coupling_utilities.py]: `list of float:`

File: lib/improver/tests/set_up_test_cubes.py [hunk 8]
```diff
@@ -109,7 +109,7 @@ def construct_scalar_time_coords(time, time_bounds, frt):
             Single forecast reference time point
 
     Returns:
-        coord_dims (list):
+        list:
```
Comment (cgsandford) [inline lib/improver/tests/set_up_test_cubes.py]: `list of tuple:`?  Not sure how best to describe this one actually!

File: lib/improver/utilities/cube_checker.py [hunk 9]
```diff
@@ -158,7 +158,7 @@ def find_dimension_coordinate_mismatch(
                 second_cube - first_cube
 
     Returns:
-        result (list):
+        list:
```
Comment (cgsandford) [inline lib/improver/utilities/cube_checker.py]: `list of str:`

File: lib/improver/utilities/cube_manipulation.py [hunk 10]
```diff
@@ -497,7 +497,7 @@ def compare_attributes(cubes, attribute_filter=None):
             A string to filter which attributes are actually compared. If None
             all attributes are compared.
     Returns:
-        unmatching_attributes (list):
+        list:
```
Comment (cgsandford) [inline lib/improver/utilities/cube_manipulation.py]: `list of dict:`

File: lib/improver/utilities/cube_manipulation.py [hunk 11]
```diff
@@ -539,7 +539,7 @@ def compare_coords(cubes):
             List of cubes to compare (must be more than 1)
 
     Returns:
-        unmatching_coords (list):
+        list:
```
Comment (cgsandford) [inline lib/improver/utilities/cube_manipulation.py]: `list of dict:`

File: lib/improver/utilities/temporal.py [hunk 12]
```diff
@@ -319,7 +319,7 @@ def set_utc_offset(longitudes):
             List of longitudes.
 
     Returns:
-        utc_offsets (list):
+        list:
```
Comment (cgsandford) [inline lib/improver/utilities/temporal.py]: `np.ndarray:`

File: lib/improver/wxcode/weather_symbols.py [hunk 13]
```diff
@@ -269,7 +269,7 @@ def create_condition_chain(self, test_conditions):
             test_conditions (dict):
                 A query from the decision tree.
         Returns:
-            condition_chain (list):
+            list:
```
Comment (cgsandford) [inline lib/improver/wxcode/weather_symbols.py]: `list of str:`

File: lib/improver/utilities/spatial.py [hunk 14]
```diff
@@ -489,7 +489,7 @@ def lat_lon_transform(trg_crs, latitude, longitude):
             Longitude coordinate.
 
     Returns:
-        x, y (float):
+        float:
             Longitude and latitude transformed into the target coordinate
             system.
```
[the remark this answers is not in the document]
Comment (neilCrosswaite) [reply lib/improver/utilities/spatial.py]:
```
    Returns:
        (tuple): tuple containing:
            **x** (float): The x coordinate in target coordinate system.
            **y** (float): The y coordinate in target coordinate system.
```

File: lib/improver/utilities/temporal.py [hunk 15]
```diff
@@ -132,7 +132,7 @@ def forecast_period_coord(
             Desired units for the resulting forecast period coordinate.
 
     Returns:
-        coord (iris.coords.AuxCoord or DimCoord):
+        iris.coords.AuxCoord or DimCoord:
```
[the remark this answers is not in the document]
Comment (neilCrosswaite) [reply lib/improver/utilities/temporal.py]: found two others.

File: lib/improver/utilities/temporal.py [hunk 16]
```diff
@@ -319,7 +319,7 @@ def set_utc_offset(longitudes):
             List of longitudes.
 
     Returns:
-        utc_offsets (list):
+        np.ndarray:
```
Comment (cgsandford) [inline lib/improver/utilities/temporal.py]: `numpy.ndarray`

**Pull request merged**